cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -736,6 +741,10 @@ private BatchPhaseSpec getPhaseSpec(String programName, PipelinePhase phase) {
       phaseConnectorDatasets.put(connectorStage.getName(), connectorDatasets.get(connectorStage.getName()));
     }
 
+    for (StageSpec alertPlusherInfo : phase.getStagesOfType(AlertPublisher.PLUGIN_TYPE)) {
```

Comment:
don't need this second loop if you create phaseConnectorDatasets before the first alert publisher loop.